doc/pages/expansions.asciidoc

```diff
@@ -32,9 +54,10 @@ Like "variable expansion" and "command substitution" in shell programming,
 Kakoune expansions can expand to multiple "words" - that is, separate
 arguments on the resulting command-line. However, unlike shell programming,
 Kakoune expansions cannot _accidentally_ expand to multiple words, even if
-the expansion contains whitespace or other special characters. While in
-shell-programming it's good practice to always wrap expansions in
-double-quotes, in Kakoune it's perfectly safe to leave expansions unquoted.
+the expansion contains whitespace or other special characters: only list types
+can expand to multiple arguments. While in shell-programming it's good practice
```

Comment:
"only list types can expand to multiple arguments" is not the reason that "Kakoune expansions cannot accidentally expand to multiple words". The reason is that Kakoune does word-splitting *before* expansion, as opposed to the shell doing it the other way around. That's true whether or not any given expansion expands to multiple arguments.

I'm not sure it's worth trying to nail down which kinds of expansions can expand to multiple words, since there's so many. List-type options are one case, but so are all `%reg{}` expansions and `%arg{@}` specifically (but not the other `%arg{}` expansions) and various other things... and explaining all that would distract from the main point of this paragraph, which is "in Kakoune it's perfectly safe to leave expansions unquoted".

If you think it's really important, I'd probably revert this change, and instead change the first couple of lines to:

> Like "variable expansion" and "command substitution" in shell programming, some Kakoune expansions (list-type options, registers, selections, etc.) can expand to multiple "words" - that is, separate...

Maybe even change "variable expansion" to "glob expansion", since shell-reparsing is usually unwanted for both variable expansion and command substitution. People *want* globs to expand to multiple words.